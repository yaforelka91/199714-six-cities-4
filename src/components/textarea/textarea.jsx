import React from 'react';
import {textareaTypes} from '../../types/types';

const Textarea = ({currentMessage, onTextareaChange}) => {
  return (
    <textarea
      className="reviews__textarea form__textarea"
      id="review" name="review"
      value={currentMessage}
      placeholder="Tell how was your stay, what you like and what can be improved"
      onChange={onTextareaChange}
    />
  );
};

Textarea.defaultProps = {
  currentMessage: ``,
};

Textarea.propTypes = textareaTypes;

export {Textarea};
export default React.memo(Textarea);
