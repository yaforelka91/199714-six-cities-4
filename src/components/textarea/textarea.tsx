import * as React from 'react';

type Props = {
  currentMessage: string;
  maxLength: number;
  onTextareaChange: () => void;
};

const Textarea: React.FC<Props> = (props: Props) => {
  const {
    currentMessage = ``,
    maxLength = 100,
    onTextareaChange
  } = props;
  return (
    <textarea
      className="reviews__textarea form__textarea"
      id="review" name="review"
      value={currentMessage}
      maxLength={maxLength}
      placeholder="Tell how was your stay, what you like and what can be improved"
      onChange={onTextareaChange}
    />
  );
};

export {Textarea};
export default React.memo(Textarea);
