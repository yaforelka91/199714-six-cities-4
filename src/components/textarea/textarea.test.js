import React from 'react';
import renderer from 'react-test-renderer';
import {Textarea} from './textarea.jsx';

describe(`TextareaSnapshot`, () => {
  it(`should render Textarea`, () => {

    const tree = renderer.create(
        <Textarea
          onTextareaChange={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
