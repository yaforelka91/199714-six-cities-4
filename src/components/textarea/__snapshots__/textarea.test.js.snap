// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaSnapshot should render Textarea 1`] = `
<textarea
  className="reviews__textarea form__textarea"
  id="review"
  name="review"
  onChange={[Function]}
  placeholder="Tell how was your stay, what you like and what can be improved"
  value=""
/>
`;
